cores/arduino/HardwareSerial.cpp

```diff
@@ -1,317 +0,0 @@
-/*
```

Comment:
With this entire file being deleted, this also deleted the GD32 HardwareSerial / UART implementation? No more `Serial`, `Serial2`, etc? (I'll check this out more closely and run this branch too to verify)

Reply:
I screwed up.

Serial needs to be ported on top of Arduino's new core ring buffer, but absolutely needs to be in there.